optimum/pipelines/pipelines_base.py

```diff
@@ -293,9 +293,28 @@ def load_ort_pipeline(
     return model, model_id, tokenizer, feature_extractor
 
 
+def load_executorch_pipeline(
+    model,
+    targeted_task,
+    load_tokenizer,
+    tokenizer,
+    feature_extractor,
+    load_feature_extractor,
+    SUPPORTED_TASKS,
+    subfolder: str = "",
+    token: Optional[Union[bool, str]] = None,
+    revision: str = "main",
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    config: AutoConfig = None,
+    **kwargs,
+):
+    raise NotImplementedError("Executorch pipeline is not implemented yet.")
+
+
 MAPPING_LOADING_FUNC = {
     "ort": load_ort_pipeline,
     "bettertransformer": load_bettertransformer,
+    "executorch": load_executorch_pipeline,
 }
```

Comment:
Is this pipeline ever used? If yes, can you show me examples?

The example in the landing page is using the transformers pipeline directly: https://github.com/huggingface/optimum?tab=readme-ov-file#run-the-exported-model-using-onnx-runtime. I fail to understand how it is possible to run the exported model with transformers pipeline. If the exported model may have a modified signature that is different from eager, which won't work with the eager `generate` I think?

Reply:
yes with the current integration executorch models will not be compatible with transformers pipelines, which is not a big deal as I don't think many people uses pipelines with optimum models

Reply:
> yes with the current integration executorch models will not be compatible with transformers pipelines, which is not a big deal as I don't think many people uses pipelines with optimum models

Okay, I think we can add pipeline support later in a separate PR. Would focus to add minimal backbone for ExecuTorch in this initial integration.